Notes/Algorithms/Data Structures/Hash-Tables.c

```diff
@@ -71,41 +78,83 @@ Item * createItem(char key[], int value, int * success) {
 // creates a new HashTable struct
 // returns new HashTable struct
 HashTable * createHashTable(int size) {
-    if (size < 1 || size > 100) return NULL;
+    if (size < 1) return NULL;
```

Comment:
Not something for cleanup, more of a passing reflection:

Using an `int` as a size necessitates this check, because `int` can hold negative values.  Is there a type that will never be negative?  Using that as a type would mean we don't even need this check, because the compiler is checking for us.

https://stackoverflow.com/questions/2550774/what-is-size-t-in-c

Reply:
Ah that's a great point -thank you for bringing this to my attention!